### PR TITLE
New Elements panel (opt-in)

### PR DESCRIPTION
**Pull request by bvaughn**
Changes in this branch:
* Add new `GenericList` component and abstract `GenericListData` class
* Update the CSS Rules panel to use these new components
* Rewrite Elements tree to use these components as well
* Add opt-in experimental settings to use new Elements tree

Some UX improvements over the current Elements tree:
* Only render the head of (iframe) `#document` nodes (old Elements panel showed `</#document>` tail)
* Selected sub-tree indicator line looks nicer
* Head nodes (e.g. `<div>` and tail nodes (e.g. `</div>`) can both be selected
* Loading errors show an error boundary (rather than being uncaught and hanging)

The legacy Elements code is deleted in #9755.

New Elements tree:
Screen Shot 2023-09-26 at 11 45 02 AM

New experimental opt-in setting:
Screen Shot 2023-09-26 at 11 46 26 AM

### CODE REVIEW

**Comment by jasonLaster** [comment]
@bvaughn looking great!

Looks like iframes might not be supported though ([replay](https://app.replay.io/recording/replay-of-devtools-git-fe-1885-part-2-recordreplayvercelapp--38961471-fb2d-4f45-846f-1ff73629b734?point=72043118931589322225584988380730714&time=26689&focusWindow=eyJiZWdpbiI6eyJwb2ludCI6IjAiLCJ0aW1lIjowfSwiZW5kIjp7InBvaW50IjoiOTU0MDg0NTQ4MDE0NjMwMDkyNDU5MjY5Mjk0OTI2MDY5NzYiLCJ0aW1lIjozNzg5NH19)) ([original replay](https://devtools-git-fe-1885-part-2-recordreplay.vercel.app/recording/cypresstestssurfaceauthcloseupp1-c2711059-td-standard-pinjs--f86ea238-5721-4b85-bfa9-f1372a353041?e2e=&apiKey=&point=15576890583921573303633942165851298&time=11316.646370023418&focusWindow=eyJiZWdpbiI6eyJwb2ludCI6IjAiLCJ0aW1lIjowfSwiZW5kIjp7InBvaW50IjoiMjEwOTM3MDU5OTk5NTEwNzcxMjAxMjA0OTA4NTMxMzg0MzIiLCJ0aW1lIjoxMzEzMX19))

**Comment by bvaughn** [comment]
Ah, good catch @jasonLaster. I bet it's because my filtering logic hides all document nodes when really it should only hide the _root_ one. I'll take a look.

**Comment by bvaughn** [comment]
Yup, that was it. Easy fix.
Screen Shot 2023-09-23 at 1 56 48 PM

**Comment by jasonLaster** [comment]
https://www.loom.com/share/e500369330864b1e85877542b80afd07

**Comment by bvaughn** [comment]
I think that _not_ showing a tail tag for document roots would be a nice change for the new Elements panel. I might try that.
Screen Shot 2023-09-23 at 2 54 12 PM

**Comment by bvaughn** [comment]
Not ignoring your comment, but I'm not going to watch a Loom for bug reports on the weekend 😁

**Review by hbenl — APPROVED** [review]
There's an empty file called `getDisplayedNodeName.ts`...

File: packages/replay-next/components/elements/constants.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  TEXT_NODE: 3,
+  CDATA_SECTION_NODE: 4,
+  ENTITY_REFERENCE_NODE: 5,
+  ENTITY_NODE: 6,
+  PROCESSING_INSTRUCTION_NODE: 7,
+  COMMENT_NODE: 8,
+  DOCUMENT_NODE: 9,
+  DOCUMENT_TYPE_NODE: 10,
+  DOCUMENT_FRAGMENT_NODE: 11,
+  NOTATION_NODE: 12,
```
[the remark this answers is not in the document]
Comment (bvaughn) [reply packages/replay-next/components/elements/constants.ts:13]: Oh I didn't know. The old Elements stuff was importing `shared/dom-node-constants.ts` (which replay-next coudln't import). Thanks

[the remark this answers is not in the document]
Comment (bvaughn) [reply packages/replay-next/components/elements/constants.ts:13]: FWIW switching over to reference that value causes a runtime error:
> TypeError: Cannot read properties of undefined (reading 'DOCUMENT_NODE')

**Pull request merged**